google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -323,15 +353,39 @@ def _create(
             request_metadata (Sequence[Tuple[str, str]]):
                 Strings which should be sent along with the create_dataset
                 request as metadata. Usually to specify special dataset config.
+            encryption_spec_key_name (Optional[str]):
+                Optional. The Cloud KMS resource identifier of the customer
+                managed encryption key used to protect the dataset. Has the
+                form:
+                ``projects/my-project/locations/my-region/keyRings/my-kr/cryptoKeys/my-key``.
+                The key needs to be in the same region as where the compute
+                resource is created.
+
+                If set, this Dataset and all sub-resources of this Dataset will be secured by this key.
 
+                Overrides encryption_spec_key_name set in aiplatform.init.
         Returns:
             operation (Operation):
                 An object representing a long-running operation.
         """
+
+        # Use provided encryption key name or else use one from global config
+        kms_key_name = (
+            encryption_spec_key_name
+            or initializer.global_config.encryption_spec_key_name
```

Comment:
Similar to the above comment. This method can potentially be called asynchronously so defaulting should happen in the synchronous caller.

This seems applicable to all changes in this PR.